scripts/release_info.py

```diff
@@ -411,6 +513,156 @@ def main():
 
     python_utils.PRINT('Done. Summary file generated in ../release_summary.md')
 
+    while True:
+        python_utils.PRINT(
+            '******************************************************')
+        python_utils.PRINT(
+            'Please update ../release_summary.md to:\n'
+            '- have a correct changelog for '
+            'updating the CHANGELOG file\n'
+            '- have a correct list of new authors and contributors to '
+            'update AUTHORS, CONTRIBUTORS and Credits section in '
+            'about-page.directive.html\n'
+            'Confirm once you are done by entering y/ye/yes.\n')
+        answer = python_utils.INPUT().lower()
+        if answer in ['y', 'ye', 'yes']:
+            break
+
+    current_release_version = branch_name[len(
+        common.RELEASE_BRANCH_NAME_PREFIX):]
+    current_date = datetime.date.today().strftime('%d %b %Y')
+    source_branch = 'develop'
+    target_branch = 'update-changelog-for-releasev%s' % current_release_version
+    # This complete code block is wrapped in try except since in case of
+    # an exception, we should ensure that the changes made to AUTHORS,
+    # CONTRIBUTORS, CHANGELOG and about-page are reverted and the branch
+    # created is deleted. This is required to avoid errors when running
+    # the script again.
+    try:
+        with python_utils.open_file(
+            '../release_summary.md', 'r') as release_summary_file:
+            release_summary_content = release_summary_file.readlines()
+
+            python_utils.PRINT('Updating Changelog...')
+            start_index = release_summary_content.index('### Changelog:\n') + 1
+            end_index = release_summary_content.index('### Commit History:\n')
+            release_version_changelog = [
+                u'v%s (%s)\n' % (current_release_version, current_date),
+                u'------------------------\n'] + release_summary_content[
+                    start_index:end_index]
+            changelog_content = []
+            with python_utils.open_file(
+                CHANGELOG_FILEPATH, 'r') as changelog_file:
+                changelog_content = changelog_file.readlines()
+            changelog_content[2:2] = release_version_changelog
+            with python_utils.open_file(
+                CHANGELOG_FILEPATH, 'w') as changelog_file:
+                for line in changelog_content:
+                    changelog_file.write(line)
+            python_utils.PRINT('Updated Changelog!')
+
+            python_utils.PRINT('Updating Authors...')
```

Comment:
Should we create methods like `update_changelog`, `update_author` etc. and make the main method more readable? (I'm fine with this structure as well.)

Reply:
I created a function to do the common handling but I feel it will be better to have separate functions for readability. I will make the updates.

Reply:
Done.